flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -153,6 +160,52 @@ public boolean scaleResource(
         return true;
     }
 
+    private void updateBiggestVertexParallelism(
+            Context context,
+            Map<JobVertexID, Map<ScalingMetric, EvaluatedScalingMetric>> evaluatedMetrics,
+            Map<JobVertexID, ScalingSummary> scalingSummaries) {
+        int taskSlotsPerTm = context.getConfiguration().get(TaskManagerOptions.NUM_TASK_SLOTS);
+
+        Map<Integer, Set<JobVertexID>> parallelismVertexMap = new HashMap<>();
+        int maxParallelism = 0;
+
+        for (Map.Entry<JobVertexID, ScalingSummary> entry : scalingSummaries.entrySet()) {
+            JobVertexID vertex = entry.getKey();
+            ScalingSummary summary = entry.getValue();
+            int newParallelism = summary.getNewParallelism();
+
+            // Update maxParallelism if a new maximum is found
+            if (newParallelism > maxParallelism) {
+                maxParallelism = newParallelism;
+            }
+
+            // Map newParallelism to JobVertexID
+            parallelismVertexMap.computeIfAbsent(newParallelism, k -> new HashSet<>()).add(vertex);
+        }
+
+        // After the loop, retrieve the JobVertexIDs with the maxParallelism value
+        Set<JobVertexID> verticesWithMaxParallelism =
+                parallelismVertexMap.getOrDefault(maxParallelism, new HashSet<>());
+
+        //  Compute the maximum parallelism for a given taskSlotsPerTm
+        var time = maxParallelism / taskSlotsPerTm;
+        if (maxParallelism > time * taskSlotsPerTm) {
+            maxParallelism = (time + 1) * taskSlotsPerTm;
+        }
+
+        final var finalMaxNewParallelism = maxParallelism;
+
+        scalingSummaries.forEach(
+                (vertexID, summary) -> {
+                    if (verticesWithMaxParallelism.contains(vertexID)) {
+                        // Here, you update the newParallelism for each matching JobVertexID
+                        summary.setNewParallelism(
+                                finalMaxNewParallelism); // Assuming setNewParallelism updates the
```

Comment:
I wonder, why only consider the vertex with the maximum parallelism? Why not +1 all the vertices?

Reply:
Good question, it's interesting, but since I have never tested this configuration in my test, I'll stay with adjustement only for the maximum in this PR :)